int32 x
int32 y
uint16 r
uint16 g
uint16 b
